### DISCUSSION TITLE
How can I persist the data of BlockSuite editors?

### DISCUSSION THREAD

**Discussion by alissa-tung** [toeverything/blocksuite#6551] [Q&A] [answered]
I noticed that there is a `@blocksuite/sync` library, but seems that there is no doc on https://block-suite.com/ .

And there is also a `packages/framework/store/src/persistence/` modules, but I have not figure out how can I persist data

I am trying to play with the BlockSuite editors. My question is, how can I implement ordinary `Ctrl/Cmd + s`, then save the contents of editors to a local store?

Seems that hotkeys are handled by https://block-suite.com/guide/event.html#handling-hotkeys

**Comment by doodlewind** [comment, marked as answer]
You can see the [vanilla-indexeddb](https://github.com/toeverything/blocksuite/tree/master/examples/vanilla-indexeddb) example about how you can save single BlockSuite doc via providers automatically - you can simply replace this IndexedDB provider with WebSocket provider. This provider-based approach saves Yjs binaries incrementally. More community provider example will be added recently.

And you can also import or export content using the [snapshot JSON](https://blocksuite.io/guide/data-synchronization.html#snapshot-api) format and save the JSON to your own API endpoints. This could be easier but doesn't support real-time collaboration.

Regarding `@blocksuite/sync`, it's designed to sync a `DocCollection` with multiple documents. The documentation is not complete and we have plan to make it more reusable, but it may take longer time to evolve it.

**Reply by alissa-tung** [reply]
thank you
